ipld/retriever_quadrant.go

```diff
@@ -92,30 +89,16 @@ func newQuadrants(dah *da.DataAvailabilityHeader) []*quadrant {
 	return quadrants
 }
 
-// index calculates index for a share in a data square slice flattened by rows.
-//
-// NOTE: The complexity of the formula below comes from:
-//   - Goal to avoid share copying
-//   - Goal to make formula generic for both rows and cols
-//   - While data square is flattened by rows only
-//
-// TODO(@Wondertan): This can be simplified by making rsmt2d working over 3D byte slice(not flattened)
-func (q *quadrant) index(rootIdx, cellIdx int) int {
-	size := len(q.roots)
-	// half square offsets, e.g. share is from Q3,
-	// so we add to index Q1+Q2
-	halfSquareOffsetCol := pow(size*2, q.source)
-	halfSquareOffsetRow := pow(size*2, q.source^1)
-	// offsets for the axis, e.g. share is from Q4.
-	// so we add to index Q3
-	offsetX := q.x * halfSquareOffsetCol * size
-	offsetY := q.y * halfSquareOffsetRow * size
-
-	rootIdx *= halfSquareOffsetRow
-	cellIdx *= halfSquareOffsetCol
-	return rootIdx + cellIdx + offsetX + offsetY
-}
-
-func pow(x, y int) int {
-	return int(math.Pow(float64(x), float64(y)))
+// pos calculates position of a share in a data square.
+func (q *quadrant) pos(rootIdx, cellIdx int) (int, int) {
```

Comment:
we should probs have a unit test for this method since an inaccuracy in it could cause panic via nil `GetCell` -> `SetCell`

Reply:
It won't panic, but even if the unit test won't fix this either. The solution is to do checks for the size. However, these rootIdx and cellIdx are bounded by squareSize gotten from DAH and it won't ever get a wrong value out of it in the privatized code, so those ifs are unnecessary

Reply:
> It won't panic, but even if the unit test won't fix this either. 

a unit test wouldn't hurt.

Reply:
> Why is GetShares exported if it's only used internally to ipld pkg?

Unrelated to this PR

Reply:
> a unit test wouldn't hurt.

Yeah, butit's useless in this case. The existing tests cover each quadrant fetching, and it works